yoyodyne/models/base.py

```diff
@@ -295,12 +323,21 @@ def predict_step(
             batch_idx (int).
 
         Returns:
-            torch.Tensor: indices of the argmax at each timestep.
+            Tuple(torch.Tensor, torch.Tensor): position 0 are the indices of 
+            the argmax at each timestep. Position 1 are the scores for each 
+            history in beam search. It will be None when using greedy.
+
         """
         predictions = self(batch)
-        # -> B x seq_len x 1.
-        greedy_predictions = self._get_predicted(predictions)
-        return greedy_predictions
+
+        if self.beam_width is not None and self.beam_width > 1:
```

Comment:
How would we get a null (`None`) beam width value in the first place? It has a sensible default (!) so it makes sense to keep it integral. Then you can get rid of the `is not None and` part here.